src/Features/Core/Portable/Completion/Providers/AbstractInternalsVisibleToCompletionProvider.cs

```diff
@@ -0,0 +1,218 @@
+                var attributeSyntaxNodeCandidate = node.Parent?.Parent?.Parent;
+                if (syntaxFactsService.IsAttribute(attributeSyntaxNodeCandidate))
+                {
+                    return attributeSyntaxNodeCandidate;
+                }
+            }
+
+            return null;
+        }
+
+        private static async Task<bool> CheckTypeInfoOfAttributeAsync(Document document, SyntaxNode attributeNode, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelForNodeAsync(attributeNode, cancellationToken).ConfigureAwait(false);
+            var typeInfo = semanticModel.GetTypeInfo(attributeNode);
+            var type = typeInfo.Type;
+            if (type == null)
+            {
+                return false;
+            }
+
+            var internalsVisibleToAttributeSymbol = semanticModel.Compilation.GetTypeByMetadataName(typeof(InternalsVisibleToAttribute).FullName);
+            return type.Equals(internalsVisibleToAttributeSymbol);
+        }
+
+        private async Task AddAssemblyCompletionItemsAsync(CompletionContext context, CancellationToken cancellationToken)
+        {
+            var currentProject = context.Document.Project;
+            var allInternalsVisibleToAttributesOfProject = await GetAllInternalsVisibleToAssemblyNamesOfProjectAsync(context, cancellationToken).ConfigureAwait(false);
+            var currentCompilation = await currentProject.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+            var currentAssemblyInfo = currentCompilation.Assembly;
+            foreach (var project in context.Document.Project.Solution.Projects)
+            {
+                if (project == currentProject)
+                {
+                    continue;
+                }
+
+                if (allInternalsVisibleToAttributesOfProject.Contains(project.AssemblyName) == true)
+                {
+                    continue;
+                }
+
+                var projectGuid = project.Id.Id.ToString();
+                var completionItem = CommonCompletionItem.Create(
+                    displayText: project.AssemblyName,
+                    rules: CompletionItemRules.Default,
+                    glyph: project.GetGlyph(),
+                    properties: ImmutableDictionary.Create<string, string>().Add(ProjectGuidKey, projectGuid));
+                context.AddItem(completionItem);
+            }
+        }
+
+        private async Task<IImmutableSet<string>> GetAllInternalsVisibleToAssemblyNamesOfProjectAsync(CompletionContext completionContext, CancellationToken cancellationToken)
+        {
+            var project = completionContext.Document.Project;
+            var resultBuilder = default(ImmutableHashSet<string>.Builder);
+            foreach (var document in project.Documents)
+            {
+                var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
@rchande 
Performance consideration:

I don't know how expensive it is to get the syntax tree of all documents of a project but in 99% of all cases `InternalVisibleTo` attributes are only defined in one document of the solution. If performance is a concern here we could replace

`foreach (var document in project.Documents) `

with

`var document = completionContext.Document`